sdl_android/src/main/java/com/smartdevicelink/api/lockscreen/LockScreenManager.java

```diff
@@ -0,0 +1,274 @@
+package com.smartdevicelink.api.lockscreen;
+
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.util.Log;
+
+import com.smartdevicelink.api.BaseSubManager;
+import com.smartdevicelink.protocol.enums.FunctionID;
+import com.smartdevicelink.proxy.RPCNotification;
+import com.smartdevicelink.proxy.interfaces.ISdl;
+import com.smartdevicelink.proxy.rpc.OnDriverDistraction;
+import com.smartdevicelink.proxy.rpc.OnHMIStatus;
+import com.smartdevicelink.proxy.rpc.OnSystemRequest;
+import com.smartdevicelink.proxy.rpc.enums.DriverDistractionState;
+import com.smartdevicelink.proxy.rpc.enums.HMILevel;
+import com.smartdevicelink.proxy.rpc.enums.LockScreenStatus;
+import com.smartdevicelink.proxy.rpc.enums.RequestType;
+import com.smartdevicelink.proxy.rpc.listeners.OnRPCNotificationListener;
+import com.smartdevicelink.util.HttpUtils;
+
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+
+/**
+ * <strong>LockscreenManager</strong> <br>
+ *
+ * Note: This class must be accessed through the SdlManager. Do not instantiate it by itself. <br>
+ *
+ * The LockscreenManager handles the logic of showing and hiding the lock screen. <br>
+ *
+ */
+public class LockScreenManager extends BaseSubManager {
+
+	private static final String TAG = "LockScreenManager";
+	private WeakReference<Context> context;
+	private HMILevel hmiLevel;
+	private boolean driverDistStatus, lockScreenEnabled, showOEMLogo;
+	private int lockScreenIcon, lockScreenColor, customView;
+	private OnRPCNotificationListener systemRequestListener, ddListener, hmiListener;
+	private String OEMIconUrl;
```

Comment:
`OEM` should be `device` per previous comment. The OEM reference should also be updated throughout the class.